### PR TITLE
Add Order model, Dao and Service for it

### PR DESCRIPTION
**Pull request by OleksiiTromsa**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Order o "
+                    + "LEFT JOIN FETCH o.tickets "
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:42]: Lesh, you need to use distinct here or when you will try to get orders, you are going to have many orders with one ticket, even if those tickets from one order.
And I am not sure but looks like inner joining is enough here.

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    + "WHERE o.user.id = :userId", Order.class)
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: No need user id just check it by o.user = :user

File: src/main/java/com/dev/cinema/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany(fetch = FetchType.LAZY)
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/model/Order.java]: @OneToMany fetch type lazy by default or not?

**Review by Nokafan — COMMENTED** [review]
Good job.

File: src/main/java/com/dev/cinema/model/Order.java (hunk 3 above)
Comment (OleksiiTromsa) [reply src/main/java/com/dev/cinema/model/Order.java]: I have already deleted it)

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (OleksiiTromsa) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: Ok

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (OleksiiTromsa) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:42]: I'm not sure, you're right, I will test.

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService cartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = orderDao.add(new Order(new ArrayList<>(tickets), user));
+        ShoppingCart cart = cartService.getByUser(user);
+        cartService.clear(cart);
+        return order;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:27]: how about orderDate?

Comment (OleksiiTromsa) [reply src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:27]: It is setted in the constructor

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**